internal/backupcontroller/etcdstrategy_controller.go

```
@@ -0,0 +1,1218 @@
+// ---------------------------------------------------------------------------
+// Validation
+// ---------------------------------------------------------------------------
+
+// validateEtcdApplicationRef rejects ApplicationRefs that name a
+// Kind/APIGroup the Etcd driver does not own. Mirrors the FoundationDB
+// driver's defence: without this gate a ref like other.example.com/Etcd
+// would be accepted by the Kind check alone and then reconciled against
+// the wrong CRD via the typed client.
+func validateEtcdApplicationRef(ref corev1.TypedLocalObjectReference) error {
+	if ref.Kind != etcdAppKind {
+		return fmt.Errorf("Etcd strategy supports applicationRef.kind=%q, got %q", etcdAppKind, ref.Kind)
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)